src/parsley/parse_to_object.py

```
@@ -0,0 +1,264 @@
+    Raises ValueError on any structural problem.
+    """
+
+    LOG_MAGIC = b'LOG'                       # ASCII “LOG” = 0x4c4f47
+    HEADER_FMT = '<IIB'                      # SID(uint32 LE), timestamp(uint32 LE), DLC(uint8)
+    HEADER_LEN = struct.calcsize(HEADER_FMT) # == 9
+    PARSE_LOGGER_PAGE_SIZE = 4096
+
+    def parse(self, buf: bytes, page_number: int) -> ParsleyObject | ParsleyError:
+        # Strip the buffer to 4096 bytes, as required by the logger.
+        if len(buf) != self.PARSE_LOGGER_PAGE_SIZE:
+            raise ValueError('Logger message must be exactly 4096 bytes')
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)